### ISSUE TITLE
Themeswitcher needed

### ISSUE THREAD

**Issue by j00ch** [devsnd/cherrymusic#605]
**Intro**
Let me start by saying that i think light themes are an unnecessary health hazard. The first computers got it right: green on black, which equals low contrast and puts less stress on the monitor and users eyes.
This is just my opinion of course but still means users should have a choice in the way CherryMusic is displayed, in other words: we need a theme switcher.

**Problem**
I created a nice dark theme somewhat like Fruity Loops, build from the compiled CSS code that comes with CherryMusic (cherrymusic.css & login.css).

![CherryMusic dark theme](https://lh3.googleusercontent.com/-FaHyZDgzjr8/Vox_RIQICWI/AAAAAAAAAFE/YjUuNDp5-lg/w426-h459/Schermafdruk%2Bvan%2B2016-01-06%2B03-34-57.png)

I then tried to implement various theme switchers in the development scenario, but they all fail due to the way CherryMusic is set up (no PHP and compiling of code)

Anyone care to give me a few pointers?

**Comment by systems-rebooter** [comment]
Wow! Your theme looks really nice!

 I've tried to play a bit with default style, but never get satisfied with results, so switched back to native CM theme, which is very cool because of its cleanness

**Comment by devsnd** [comment]
Hey Jochum,

Thanks for the input. I would welcome a new darker theme for CherryMusic; Or at least to give it out as an optional hack for people to use. If you want to change the underlying CSS of cherrymusic you can enter the client developer mode, by adding the devel=True parameter (this enables the LESS compiler on page load, thats why it will take longer to load)

    http://localhost:8080/?devel=True

Assuming you run CM locally on port 8080.

Now you can edit the LESS files which will be recompiled every time you reload the page. So it's best to start editing the files here in [/res/bootstrap3/less](https://github.com/devsnd/cherrymusic/tree/devel/res/bootstrap3/less). Most importantly, look at the ```variables.less``` file, which allows you to change a lot of colors in at once.

You can [read more about LESS here](http://lesscss.org/)

**Comment by j00ch** [comment]
@systems-rebooter : Glad you like it, hopefully i'll be able to implement it soon.
@devsnd : 
I was able to find the developer mode, but implementing CSS & JS code after LESS does not work. I have worked a little bit with LESS in the past, but nothing in the magnitude of multiple themes.
My initial plan was to edit the LESS files but it turned out to be easier to just edit one big CSS file (cherrymusic.css) by hacking CM in a browser with Firefox's development tools.

Anyways thanks for the pointer to variables.less, that will help me convert my CSS hack into LESS code.

I'm currently looking into LESS specific theme switchers. Hope this one will work: http://stevedrivendevelopment.com/2014/06/20/dotless-namedthemes-add-file-based-theme-swithing-in-less-css-preprocessing/
